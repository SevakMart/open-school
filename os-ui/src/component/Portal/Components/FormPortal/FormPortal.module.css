.overlayStyles {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0.9;
  background-color: #222427;
  z-index: 100;
}

.modalStyles {
  position: fixed;
  top: 50%;
  left: 50%;
  width: 40%;
  height: 90%;
  background-color: #ffffff;
  transform: translate(-50%, -50%);
  overflow-y: auto;
  scrollbar-width: none;
  border-radius: 40px;
  z-index: 100;
}
.modalStyles::-webkit-scrollbar {
  display: none;
}/*# sourceMappingURL=FormPortal.module.css.map */