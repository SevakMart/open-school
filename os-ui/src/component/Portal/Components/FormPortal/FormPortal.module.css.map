{"version":3,"sources":["FormPortal.module.scss","../../../../styles/Configs/Colors.scss","FormPortal.module.css"],"names":[],"mappings":"AAEA;EACI,eAAA;EACA,MAAA;EACA,QAAA;EACA,SAAA;EACA,OAAA;EACA,YAAA;EACA,yBCNgB;EDOhB,YAAA;AEDJ;;AFGA;EACI,eAAA;EACA,QAAA;EACA,SAAA;EACA,UAAA;EACA,WAAA;EACA,yBCde;EDef,gCAAA;EACA,gBAAA;EACA,qBAAA;EACA,mBAAA;EACA,YAAA;AEAJ;AFCI;EACI,aAAA;AECR","file":"FormPortal.module.css"}