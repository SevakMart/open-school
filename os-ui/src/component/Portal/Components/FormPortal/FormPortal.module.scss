@import "../../../../styles/Configs/Colors.scss";

.overlayStyles {
    position:fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    opacity: 0.9;
    background-color: $header-main-color;
    z-index: 100;
}
.modalStyles {
    position:fixed;
    top:50%;
    left:50%;
    width:40%;
    padding-bottom: 20px;
    max-height: 90%;
    background-color: $white-text-color;
    transform:translate(-50%,-50%);
    overflow-y: auto;
    scrollbar-width: none;
    border-radius: 40px;
    z-index:100;
    &::-webkit-scrollbar {
        display: none;
    }
}

.enrolledPopup {
    position:fixed;
    top:50%;
    left:50%;
    background-color: $white-text-color;
    transform:translate(-50%,-50%);
    overflow-y: auto;
    scrollbar-width: none;
    border-radius: 40px;
    z-index:100;
    &::-webkit-scrollbar {
        display: none;
    }
}