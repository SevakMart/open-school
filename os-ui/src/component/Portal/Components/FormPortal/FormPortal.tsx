import { createPortal } from 'react-dom';
import styles from './FormPortal.module.scss';

const portalElement = document.createElement('div');
portalElement.setAttribute('id', 'formPortal');
document.body.appendChild(portalElement);

export const FormPortal = ({ isOpen, children }:{isOpen:boolean, children:any}) => {
  if (!isOpen) return null;
  const { overlayStyles, modalStyles } = styles;

  return createPortal(
    <>
      <div className={overlayStyles} />
      <div className={modalStyles}>
        {children}
      </div>
    </>,
    portalElement,
  );
};
