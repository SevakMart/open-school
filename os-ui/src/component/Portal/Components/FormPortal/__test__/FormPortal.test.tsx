import { render } from '@testing-library/react';
import { FormPortal } from '../FormPortal';

describe('Create unit tests for FormPortal component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(<FormPortal isOpen children={<h2>Hello world</h2>} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
