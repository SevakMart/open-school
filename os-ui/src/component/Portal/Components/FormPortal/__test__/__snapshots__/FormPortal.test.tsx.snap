// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create unit tests for FormPortal component Create a snapshot test 1`] = `<DocumentFragment />`;
